api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -333,6 +333,15 @@ default UpdateStatistics updateStatistics() {
    */
   Map<String, SnapshotRef> refs();
 
+  /**
+   * Returns the UUID of the table
+   *
+   * @return the UUID of the table
+   */
+  default String uuid() {
```

Comment:
Why String type?

Reply:
I would use `UUID` but internally we are using `String`, so it looks correct here (or we should use `UUID` internally, but it's a different story 😄 ).

Reply:
Yeah...currently it's a string. In Python we are using actual UUIDs but for the Java side I don't think we can change this :(

Reply:
We don't necessarily have to make the public API match our internal type. We could parse the string into a UUID if we chose to. I think it's just a String since we'll convert it to/from string for the metadata JSON anyway.

Reply:
That's a good point, we don't need to stick to the internal types on the API. If I think about what's best here from an API perspective, it would be to surface a UUID type. 
From a simple API semantic perspective, for someone unfamiliar with all of this, it's probably just makes more sense that `table.uuid()` returns an actual java.util.UUID.

Also if one wants to store this data in external systems and leverage their UUID data types (which would be more compact/efficient for lookups, although probably negligible in practice) then the API avoids the extra conversion burden on the user. I think this advantage is probably minimal/negligible though.

Reply:
Updated the API to be an actual UUID